### ISSUE TITLE
Improve command history search by limiting the history items size.

### ISSUE THREAD

**Issue by CfirTsabari** [warpdotdev/Warp#1892]
### Describe the solution you'd like?

```text
When searching the commands history exclude items that are bigger than 10000 chars.
```

### Is your feature request related to a problem? Please describe.

```text
If the command history file contains very large items the performance of the search is very poor.
Might take 10 seconds to type one char.
I had this issue and I solved it by removing the large items.
```

### How important is this feature to you?

3

**Comment by elviskahoro** [comment]
What do you mean exactly by large items? Very long commands? Multi-line commands?

**Comment by dannyneira** [comment]
Thanks for this feature request! I suspect this means long commands, like an echo with 10k+ characters. Please +1 this request as it helps us gauge interest. We will post any updates here.

**Comment by CfirTsabari** [comment]
Yes i mean long command, of course this commands are usually accedintly copy-paste commands so there is not point in using them, I am pretty sure `fzf` ignores these commands.

**Comment by varenc** [comment]
I too would like to report that Warp history search is basically unusable for me due to it's very slow speed. Each keypress in the history search box takes a couple seconds to appear and very confusingly the keystrokes are queues up so after typing a few I just sit and watch them slowly appear. 

At the minimum Warp could improve performance by cancelling those costly searches when other keystrokes are in the queue and only search when I've stopped typing.  While I'm waiting it for all my keystrokes to appear it doesn't make sense to run slow searches that'll soon be discarded.  Beyond that, maybe Warp could keep an in-memory copy of the history in an instantly substring searchable data structure and just periodically update the state of history in the background.

Regarding my history size, I'm not sure if it's strictly just due to large/long history items like this issue says. It might just be the vast size of my history file.

```bash
cat ~/.zsh_history | wc
  183686 1435253 17457729
```

So 183k lines averaging 7.81 words per line.  Though lines don't map 1:1 with history items, due to multi-line commands, and I actually have 154k items in my history.

My history file is certainly unusually large, but it's super useful to me. And I have no problem using a vanilla terminal and either zsh's built in history searching, or the fzf history searching I use regularly. With fzf searching there's a slightly delay as all the items are pulled in but the most immediate items are shown immediately. And after pulling in all the history all searches on it are immediate.
